### PR TITLE
balancedParentheses.js

### PR DESCRIPTION
**Pull request by ganeshyevle**
Given a string str of length N, consisting of ‘(‘ and ‘)‘ only, the task is to check whether it is balanced or not. Examples: 

Input: str = “((()))()()”  

Output: Balanced

 Input: str = “())((())”  

Output: Not Balanced

### CODE REVIEW

File: DSA_Training/balancedParentheses.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+            if (stack.length === 0) {
+                return "Not Balanced"; // Unmatched closing parenthesis
+            }
+            stack.pop(); // Matched closing parenthesis, pop from stack
+        }
+    }
+
+    return stack.length === 0 ? "Balanced" : "Not Balanced"; // Check if stack is empty
+}
+
```
Comment (Umarfarook1) [inline DSA_Training/balancedParentheses.js:21]: Remove predefined inputs and add user input

File: DSA_Training/balancedParentheses.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+function isBalanced(expression) {
```
Comment (Umarfarook1) [inline DSA_Training/balancedParentheses.js:2]: please add comment here

File: DSA_Training/balancedParentheses.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    const stack = []; // Stack to track opening parentheses
+
```
Comment (Umarfarook1) [inline DSA_Training/balancedParentheses.js:4]: please add comment here

File: DSA_Training/balancedParentheses.js (hunk 1 above)
Comment (Umarfarook1) [inline DSA_Training/balancedParentheses.js:21]: please add comment here,
Remove predefined inputs and add user input

**Review by Umarfarook1 — CHANGES_REQUESTED** [review]
Please go through the requested changes

File: DSA_Training/balancedParentheses.js (hunk 1 above)
Comment (ganeshyevle) [reply DSA_Training/balancedParentheses.js:21]: done

File: DSA_Training/balancedParentheses.js (hunk 2 above)
Comment (ganeshyevle) [reply DSA_Training/balancedParentheses.js:2]: done

File: DSA_Training/balancedParentheses.js (hunk 3 above)
Comment (ganeshyevle) [reply DSA_Training/balancedParentheses.js:4]: done

File: DSA_Training/balancedParentheses.js (hunk 1 above)
Comment (ganeshyevle) [reply DSA_Training/balancedParentheses.js:21]: done

**Review by Umarfarook1 — APPROVED** [review]

**Pull request merged**